blaze-core/src/main/scala/org/http4s/blaze/Http1Stage.scala

```diff
@@ -98,15 +104,13 @@ trait Http1Stage { self: TailStage[ByteBuffer] =>
           new CachingStaticWriter(rr, this) // will cache for a bit, then signal close if the body is long
         }
       }
-      else bodyEncoding match { // HTTP >= 1.1 request without length. Will use a chunked encoder
+      else bodyEncoding match { // HTTP >= 1.1 request without length and/or with chunked encoder
         case Some(enc) => // Signaling chunked means flush every chunk
-          if (enc.hasChunked) new ChunkProcessWriter(rr, this, trailer)
-          else  {   // going to do identity
-            logger.warn(s"Unknown transfer encoding: '${enc.value}'. Stripping header.")
-            rr << "\r\n"
-            val b = ByteBuffer.wrap(rr.result().getBytes(StandardCharsets.ISO_8859_1))
-            new IdentityWriter(b, -1, this)
-          }
+          bodyEncoding.filter(_.hasChunked).foreach(enc => logger.warn(s"Unsupported transfer encoding: '${enc.value}' for HTTP 1.$minor. Stripping header."))
+
+          lengthHeader.foreach(_ => logger.warn(s"Both Content-Length and Transfer-Encoding headers defined. Stripping Content-Length."))
```

Comment:
This check is unnecessary: if we hit this case we know its `None` so this is just an unnecessary lambda allocation.